docs/DeveloperGuide.md

```diff
@@ -1,15 +1,241 @@
 # Developer Guide
+This developer guide will help you to get started with our product, badMaths!
 
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements) 
+2. [Design & implementation](#design--implementation)
+   - [Store Notes](#store-notes) 
+   - [Graph](#graph)
+   - [Matrix](#matrix)
+2. [Product Scope](#product-scope)
+   - [Target user profile](#target-user-profile)
+   - [Value proposition](#value-proposition)
+3. [User Stories](#user-stories)
+4. [Non-Functional Requirements](#non-functional-requirements)
+5. [Glossary](#glossary)
+6. [Instructions for manual testing](#instructions-for-manual-testing)
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+
 ## Design & implementation
+### Solving Quadratic Equations
+The Quadratic equation solver allows user to find the value of `x` in a simple quadratic equation.
+The function is started when `solveQuadratic()` is called in the Command class.
+Below is an  example of how it would be called.
+```
+Quadratic. 2x^2 + 3x + 1
+```
+Step 1. Parser class would split the user input into 2 parts, command and toDo. Command would be `Quadratic`,
+thus `executeCommand` would trigger `quadraticSolver`. toDo would be `2x^2 + 3x + 1`.
+
+Step 2. The general formula of quadratic equations is `ax^2 + bx + c`, thus `findA`, `findB`, `findC`
+would identify a, b and c from toDo.
+
+Step 3. To solve the quadratic equation, `quadraticFormula` is called with a, b and c as parameters. The output would be
+a ArrayList of Doubles that contain 2 values, the possible values of x.
+
+Step 4. `printAnswer` is then called which would call upon the `UI` class's method: `printQuadraticAnswer`. This would output
+the two x values.
+
+Step 5. If any exceptions are caught in the above steps, `printQuadraticFormulaError` would be called from UI to show an error message
+to the user.
+
+![img_3.png](img_3.png)
+
+### Store and Display Notes
+The Store and Display Notes functionality allows users to input notes 
+which will be stored in a list, and to display a list of all notes
+stored by users. This functionality is achieved through the `Store.` and `List.` commands.
+
+![img_2.png](img_2.png)
```

Comment:
I think getCommand and getToDo for the DG at the top, the arrow should be the other way round cause it is badmaths that is calling parser's function, and the calls should have a separate activation bar